articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,196 @@
+:::
+
+![Auth0 Universal Login](https://cdn.auth0.com/blog/universal-login/lightweight-login.png)
+
+:::note
+The Google social provider is enabled by default on new tenants. Auth0 offers you developer keys to test logging in with a [social identity provider](https://auth0.com/docs/connections/identity-providers-social). However, these developer keys have some limitations that may cause your app to behave differently. For more details on this and on how to fix it, read on [Test Social Connections with Auth0 Developer Keys](https://auth0.com/docs/connections/social/devkeys#limitations-of-developer-keys).
+:::
+
+## Step 4: Log Out of Your App
+
+Now that you can log in to your app, you also need a way to log out. You can create a logout button as follows:
+
+```javascript
+import React from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+
+const LogoutButton = () => {
+  const { logout } = useAuth0();
+
+  return <button onClick={() => logout()}>Log Out</button>;
+};
+
+export default LogoutButton;
+```
+
+Then, you can render the `LoginButton` and `LogoutButton` conditionally by using the `isAuthenticated` property once again:
+
+```javascript
+import React from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+import LoginButton from "./LoginButton";
+import LogoutButton from "./LogoutButton";
+
+function App() {
+  const { isLoading, isAuthenticated } = useAuth0();
+
+  return (
+    <div className="App">
+      {isLoading ? (
+        <h2>Loading...</h2>
+      ) : (
+        <nav>{isAuthenticated ? <LogoutButton /> : <LoginButton />}</nav>
+      )}
+    </div>
+  );
+}
+
+export default App;
+```
+
+If Auth0 has authenticated the user, render the logout button; otherwise, render the login button.
+
+:::panel Checkpoint
+Add the `LogoutButton` button to your application. When you click it, verify that your React application redirects you the address you specified as one of the "Allowed Logout URLs".
+:::
+
+## Step 5: Get User Profile Information
+
+The Auth0 React SDK helps you retrieve the profile information associated with logged-in users quickly in whatever component you need it, such as their name or profile picture, to personalize a page. The profile information is available through the `user` object exposed by the `useAuth0()` custom React Hook. Take this `Profile` component as an example:
+
+```javascript
+import React from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+
+const Profile = () => {
+  const { user } = useAuth0();
+
+  return (
+    <div>
+      <img src={user.picture} alt={user.name} />
+      <h2>{user.name}</h2>
+      <p>{user.email}</p>
+    </div>
```

Comment:
If we wanted to demonstrate the use case of `isAuthenticated`, let's do it here and make sure we also link off to the docs on GitHub where more detail about that property (and others) can be found.